resources/rgbscripts/plasma.js

```diff
@@ -61,30 +61,25 @@ var testAlgo;
       if (_preset === "Rainbow")
       {
         algo.presetIndex = 0;
-        util.colorArray = [ 0xFF0000, 0x00FF00, 0x0000FF ];
       }
       else if (_preset === "Fire")
       {
         algo.presetIndex = 1;
-        util.colorArray = [ 0xFFFF00, 0xFF0000, 0x000040, 0xFF0000 ];
       }
       else if (_preset === "Abstract")
       {
         algo.presetIndex = 2;
-        util.colorArray = [ 0x5571FF, 0x00FFFF, 0xFF00FF, 0xFFFF00 ];
       }
       else if (_preset === "Ocean")
       {
         algo.presetIndex = 3;
-        util.colorArray = [ 0x003AB9, 0x02EAFF ];
       }
       else if (_preset === "User Defined")
       {
         algo.presetIndex = 4;
         algo.acceptColors = 5;
-        util.colorArray = [ 0x00FF00, 0xFFAA00, 0x0000FF, 0xFFFF00, 0xFFFFFF ];
```

Comment:
Don't reset the colors when the `presetIndex` property is set by QLC+. That property is always set at start, then colors get loaded, then colors get set. So when you reset colors here, _those_ colors are set again. Ergo: you loose the user defined colors.